### PR TITLE
Refactoring bookmark

### PR DESCRIPTION
**Pull request by Anita77777**

### CODE REVIEW

**Comment by Anita77777** [comment]
Hi Dominik, 

with help of Anita I refactored the bookmark buttons and handle the state via Zustand now. Please check and if ok approve.

File: src/components/Button/BookmarkButton.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import useStore from '../useStore/useStore';
+import { WrapperBookmark } from '../Form/Form.styled';
+import MySVG from '../SVG/SVG';
+
+export default function Bookmark() {
+	const updateBookmark = useStore(state => state.updateBookmark);
+	const bookmarkStatus = useStore(state => state.bookmarkStatus);
+	console.log(bookmarkStatus);
```
Comment (doemser) [inline src/components/Button/BookmarkButton.js]: I think you can remove that console.log.

File: src/components/Button/BookmarkButton.js [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+			<p>Tbr</p>
+			{bookmarkStatus === 'tbr' ? (
+				<div onClick={() => updateBookmark('')}>
+					<MySVG variant="bookFilled" color="#694370" />{' '}
+				</div>
+			) : (
+				<div onClick={() => updateBookmark('tbr')}>
+					<MySVG variant="bookEmpty" />
+				</div>
+			)}
```
Comment (doemser) [inline src/components/Button/BookmarkButton.js]: Something you should definitely change is having divs with an onClick, I think you should use buttons (radio-buttons maybe)for this, due to them being accessible by tabbing on your keyboard.

File: src/components/Form/Form.js [hunk 3]
```diff
@@ -15,15 +17,14 @@ export default function Form() {
 	} = useForm();
 
 	const onSubmit = book => {
-		addNewBook(book);
+		const newBook = {
+			...book,
+			bookmarkStatus: bookmarkStatus,
+		};
+		addNewBook(newBook);
```
Comment (doemser) [inline src/components/Form/Form.js]: you could just fill it like this, or is there any reason to use an extra namespace for that? :

```
		addNewBook({
			...book,
			bookmarkStatus: bookmarkStatus,
		});
```

then optimize a little:

```
		addNewBook({
			...book,
			bookmarkStatus,
		});
```

File: src/components/useStore/useStore.js [hunk 4]
```diff
@@ -27,6 +28,10 @@ const useStore = create(
 				};
 			});
 		},
+		bookmarkStatus: '',
+		updateBookmark: status => {
+			set({ bookmarkStatus: status });
+		},
```
Comment (doemser) [inline src/components/useStore/useStore.js]: set bookmarkStatus initial to `null` and also use `null` when you want to reset the state like this: `updateBookmark(null)`. This resolves bookmarkStatus to be `truthy` or `falsy` which seems to fit better to the states purpose.

**Review by doemser — CHANGES_REQUESTED** [review]
Very good work! :) I have some minor changes to request but also a bigger one.

Using divs as buttons is a major a11y problem, this should be changed. Make sure you can reach every input by tabbing on your keyboard :)

File: src/components/useStore/useStore.js (hunk 4 above)
Comment (Anita77777) [reply src/components/useStore/useStore.js]: I'm not sure I understand the last two comments - maybe you can explain to me in person tomorrow?

Comment (doemser) [reply src/components/useStore/useStore.js]: Yes, lets meet after our Daily :)

**Comment by Anita77777** [comment]
Here are the changes, added a tiny bit of styling for the buttons

**Review by doemser — APPROVED** [review]
We already talked about it. Now your good to go :)

**Pull request merged**